### PR TITLE
Escape quotes

### LINKED ISSUE #34: Add apostrophe support
image

Ran into a small bump in the road attempting to pull when using apostrophes in Contentful schemas. 

Could we use backticks as opposed to apostrophes in auto generated `.d.ts` file? This would allow for apostrophes in the schema.

### PR DESCRIPTION
**Pull request by Jaden-Giordano**
This aims to fix #34.

Prettier, in some cases, is annoyingly opinionated and doesnt seem to want to preserve escaped quotes and instead just switch to use double quotes or single quotes if the string contains one or the other (this seems like its going to be the case forever (https://github.com/prettier/prettier/issues/973). In order for prettier to correctly decide this we need to escape them inside strings.

In contentful an array of strings that contain apostrophe's will break the output: 
```ts
  tags: (
    | 'Something'
    | 'Banana'
    | 'Apples'
    | 'Someone's pear' // This line here breaks as the quote isnt escaped so prettier doesnt reformat this to correct code.
    | 'Grape'
  )[];
```
Where as if the quote is escaped it properly reformats to:
```ts
  tags: (
    | 'Something'
    | 'Banana'
    | 'Apples'
    | "Someone's pear" // Switches to double quotes
    | 'Grape'
  )[];
```

This PR specifically adds the `escapeSingleQuote`  function to a `utils.ts` in the renderer directory. It then uses this function to escape apostrophes used in contentful data (specifically text) that we drop in.

### CODE REVIEW

**Comment by Jaden-Giordano** [comment]
@lorcan Bumping this for a review.

File: src/renderers/contentful/renderContentType.ts [hunk 1]
```diff
@@ -67,7 +68,7 @@ function renderSys(sys: Sys) {
       locale: string;
       contentType: {
         sys: {
-          id: '${sys.id}';
+          id: '${escapeSingleQuotes(sys.id)}';
```
Comment (GabrielAnca) [inline src/renderers/contentful/renderContentType.ts:71]: AFAIK `sys.contentType.sys.id` is the api identifier for the content model, which can only have letters and numbers - so escaping might not be necessary here:

image

I might be wrong though 😅 Let me know what you think

File: src/renderers/contentful/renderDefaultLocale.ts [hunk 2]
```diff
@@ -7,5 +8,5 @@ export default function renderDefaultLocale(locales: Locale[]): string {
     throw new Error("Could not find a default locale in Contentful.")
   }
 
-  return `export type CONTENTFUL_DEFAULT_LOCALE_CODE = '${defaultLocale.code}';`
+  return `export type CONTENTFUL_DEFAULT_LOCALE_CODE = '${escapeSingleQuotes(defaultLocale.code)}';`
```
Comment (GabrielAnca) [inline src/renderers/contentful/renderDefaultLocale.ts:11]: Same as above, is there a need to escape the locale code? I'd expect locale codes to be only letters and dashes

**Review by GabrielAnca — COMMENTED** [review]
Thanks for this PR @Jaden-Giordano! It looks great to me! I have a couple of questions there, mainly if we need to do this for locales and `sys.contentType.sys.id` 😃 Let me know what you think!

**Comment by GabrielAnca** [comment]
I'm also investigating why the CircleCI build didn't kick off 😅

File: src/renderers/contentful/renderContentType.ts (hunk 1 above)
Comment (Jaden-Giordano) [reply src/renderers/contentful/renderContentType.ts:71]: Yea this doesnt need to be escaped I kinda just went through and find and replaced, lemme remove it from this.

File: src/renderers/contentful/renderAllLocales.ts [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import renderUnion from "../typescript/renderUnion"
 import { Locale } from "contentful"
+import { escapeSingleQuotes } from "../utils"
 
 export default function renderAllLocales(locales: Locale[]): string {
-  return renderUnion("LOCALE_CODE", locales.map(locale => `'${locale.code}'`))
+  return renderUnion(
+    "LOCALE_CODE",
+    locales.map(locale => `'${escapeSingleQuotes(locale.code)}'`),
```
Comment (GabrielAnca) [inline src/renderers/contentful/renderAllLocales.ts:8]: Sorry, I missed this one too 😅 Do you mind reverting this file too?

File: src/renderers/contentful/renderContentType.ts (hunk 1 above)
Comment (GabrielAnca) [reply src/renderers/contentful/renderContentType.ts:71]: Perfect! Looks great! Thanks @Jaden-Giordano!

File: src/renderers/contentful/renderAllLocales.ts (hunk 3 above)
Comment (Jaden-Giordano) [reply src/renderers/contentful/renderAllLocales.ts]: Cool got it, I found one more sys.id escape too and i removed the escape for that too.

File: src/renderers/contentful/renderContentType.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { ContentType, Field, FieldType, Sys } from "contentful"
 
+import { escapeSingleQuotes } from "../utils"
```
Comment (GabrielAnca) [inline src/renderers/contentful/renderContentType.ts:3]: We don't need this import anymore 😅

```suggestion
```

**Review by GabrielAnca — COMMENTED** [review]
Looks perfect, just one tiny unused import! I'm still actively working on the CircleCI issue. Hope to get it fixed during this week.

**Review by GabrielAnca — COMMENTED** [review]
Looking great! The CircleCI support said the issue should be sorted so if you make an empty push (or a rebase) it should kick off the CI build

**Comment by GabrielAnca** [comment]
Rebased the PR myself and now the build is green ✔️ 

I'm working on a change in our NPM config and I'll merge this PR as soon as that's ready, hopefully before the end of the week.

File: src/renderers/contentful/renderAllLocales.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
 import renderUnion from "../typescript/renderUnion"
 import { Locale } from "contentful"
+import { escapeSingleQuotes } from "../utils"
```
Comment (GabrielAnca) [inline src/renderers/contentful/renderAllLocales.ts:3]:
```suggestion
```

**Review by GabrielAnca — APPROVED** [review]
All green, looks perfect 👌🏻 Thanks a lot for the contribution @Jaden-Giordano

**Pull request merged**